src/coreclr/nativeaot/Runtime/riscv64/PInvoke.S

```diff
@@ -19,31 +19,30 @@
 NESTED_ENTRY RhpPInvoke, _TEXT, NoHandler
         sd    fp, OFFSETOF__PInvokeTransitionFrame__m_FramePointer(a0)
         sd    ra, OFFSETOF__PInvokeTransitionFrame__m_RIP(a0)
-        sd    sp, OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs(a0)
+        sd    t0, OFFSETOF__PInvokeTransitionFrame__m_PreservedRegs(a0)
         li    t0, PTFF_SAVE_SP
         sd    t0, OFFSETOF__PInvokeTransitionFrame__m_Flags(a0)
 
         // get TLS global variable address
-#ifdef FEATURE_EMULATED_TLS
-        GETTHREAD_ETLS_1
-#else
-        INLINE_GETTHREAD a1
-#endif
+        INLINE_GETTHREAD a1, t2
 
         sd    a1, OFFSETOF__PInvokeTransitionFrame__m_pThread(a0)
         sd    a0, OFFSETOF__Thread__m_pTransitionFrame(a1)
-        jr  ra
+        ret
 NESTED_END RhpPInvoke, _TEXT
 
 LEAF_ENTRY RhpPInvokeReturn, _TEXT
         ld    t0, OFFSETOF__PInvokeTransitionFrame__m_pThread(a0)
-        sd    zero, OFFSETOF__Thread__m_pTransitionFrame(t0)
+        mv    t1, zero
+        sd    t1, OFFSETOF__Thread__m_pTransitionFrame(t0)
 
-        PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, a5
+        PREPARE_EXTERNAL_VAR_INDIRECT_W RhpTrapThreads, t0
+
+        bnez    t1, 0f  // If TrapThreadsFlags_None is non-zero, branch
+        ret
+
+0:
+        // Passing transition frame pointer in a0
+        tail    RhpWaitForGC2  // Call RhpWaitForGC2 function
```

Comment:
```suggestion
        tail    RhpWaitForGC2
```
This comment is not very useful and it is slightly incorrect (it is a tail call).